File: app/src/main/java/com/vpulse/ftpnext/core/FTPType.java
package com.vpulse.ftpnext.core;

public enum FTPType {

    FTP(0),

    SFTP(1),

//    FTPES(2),

    DEFAULT(0);

    private final int mValue;

    FTPType(int iValue) {
        this.mValue = iValue;
    }

    public int getValue() {
        return mValue;
    }

    public static FTPType getValue(int iValue) {
        if (iValue == 0)
            return FTPType.FTP;
        else if (iValue == 1)
            return FTPType.SFTP;
//        else if (iValue == 2)
//            return FTPType.FTPES;
        return FTPType.DEFAULT;
    }
}

File: app/src/main/java/com/vpulse/ftpnext/core/FTPType.kt
package com.vpulse.ftpnext.core

enum class FTPType(val value: Int) {
    FTP(0),
    SFTP(1),  //    FTPES(2),
    DEFAULT(0);

    companion object {
        fun getValue(iValue: Int): FTPType {
            if (iValue == 0) return FTP else if (iValue == 1) return SFTP //        else if (iValue == 2)
            //            return FTPType.FTPES;
            return DEFAULT
        }
    }
}